### ISSUE TITLE
[BUG] [Ruby Client][Windows]Error is shown in terminal when writing a large amounts of log info data in log file.

### ISSUE THREAD

**Issue by lyiyu66** [vmware/singleton#1730]
**Describe the bug**
Windows only: Error is shown in terminal when writing a large amounts of log info data in log file.
Error as below: 
```
log writing failed. closed stream
log shifting failed. closed stream
log writing failed. closed stream
log shifting failed. Permission denied @ rb_file_s_rename - (./sgtnclient_d.log, ./sgtnclient_d.log.0)
log writing failed. closed stream
log shifting failed. closed stream
```

**To Reproduce**
1. Execute around 100 cases one by one in sandbox mode.
2. That will generate a large amounts of log info data to log file and it is stored in 2 log files: sgtnclient_d.log & sgtnclient_d.log.0
3. Repeat step 1
4. There are 2 log files: sgtnclient_d.log & sgtnclient_d.log.2

**Expected behavior**
No error pops up and log info can be stored in log file in order.

**Comment by Xiaochao8** [comment]
Please remember to test log Rolling as keeping 4 files and 1M each file.

**Comment by lyiyu66** [comment]
The issue is blocked by https://github.com/vmware/singleton/issues/1794 since only 1 log file is generated.

**Comment by lyiyu66** [comment]
The issue is fixed with commit 63145ec8b9f3d47d59ea3b623ce1a1cf21826b7e.
